packages/bento-design-system/src/DateField/createDateField.tsx

```diff
@@ -0,0 +1,263 @@
+
+        if (props.type === "range") {
+          props.onChange([startDate, endDate]);
+        } else {
+          props.onChange(startDate);
+        }
+      },
+      startDate: props.type === "range" ? props.value[0] : props.value,
+      endDate: props.type === "range" ? props.value[1] : null,
+      focusedInput,
+      numberOfMonths: 1,
+      minBookingDate: props.minDate,
+      maxBookingDate: props.maxDate,
+    });
+
+    const selectDate = (date: Date) => {
+      if (!isDateBlocked(date)) {
+        onDateSelect(date);
+        onDateFocus(date);
+      }
+    };
+
+    const { labelProps, descriptionProps, errorMessageProps } = useField({
+      ...props,
+    });
+
+    const onDateClear = () => {
+      if (props.type === "range") {
+        if (focusedInput === "startDate") {
+          props.onChange([null, props.value[1]]);
+        } else {
+          props.onChange([props.value[0], null]);
+        }
+      } else {
+        props.onChange(null);
+      }
+    };
+
+    const onInputFocus = (input: NonNullable<FocusedInput>) => () => {
+      if (!props.disabled && !props.readOnly) {
+        setIsFocused(true);
+        setFocusedInput(input);
+      }
+    };
+
+    const onInputClick = (input: NonNullable<FocusedInput>) => () => {
+      if (!props.disabled && !props.readOnly) {
+        setIsFocused(true);
+        setFocusedInput(input);
+        setIsOpen(true);
+      }
+    };
+
+    const shortcuts = props.shortcuts && (
+      <Inline space={4}>
+        {props.shortcuts.map((shortcut) => (
+          <Button
+            kind="transparent"
+            hierarchy="secondary"
+            size="small"
+            label={shortcut.label}
+            onPress={() => {
+              if (props.type === "range") {
+                const value = shortcut.value as [Date, Date];
+                props.onChange(value);
+                onDateFocus(value[0]);
+              } else {
+                const value = shortcut.value as Date;
+                props.onChange(value);
+                onDateFocus(value);
+              }
+            }}
+          />
+        ))}
+      </Inline>
+    );
+
+    return (
+      <Field
+        {...props}
+        labelProps={labelProps}
+        assistiveTextProps={descriptionProps}
+        errorMessageProps={errorMessageProps}
+      >
+        <Box
+          borderRadius={inputConfig.radius}
+          paddingX={inputConfig.paddingX}
+          paddingY={inputConfig.paddingY}
+          className={clsx(
+            inputRecipe({ validation: validationState || "notSet" }),
+            bodyRecipe({
+              color: props.disabled ? "disabled" : "default",
+              weight: "default",
+              size: inputConfig.fontSize,
+            }),
+            dateFieldRecipe({ validation: validationState || "notSet", isFocused }),
+            {
+              readOnly: props.readOnly,
```

Comment:
questo cos'è? Una classe che si chiama `readOnly`? Perché hardcoded così?

Reply:
vedi:
https://github.com/buildo/bento-design-system/pull/131/files#diff-36faa0e45adf364df9b0beb158835c417a52cba14309bbcdb2034228b63bfc56

la pseudo-class `:read-only` viene sempre applicata ai `div`, quindi in questo caso avremmo l'input "fittizio" sempre in stato di readOnly. Per risolvere, ho ristretto la condizione che usa la pseudo-class ai soli elementi `input`, ma ho quindi bisogno di un modo per attivarla condizionalmente anche ai `div` quando il field è veramente in readOnly. L'alternativa è averla come variant di `inputRecipe` e passarla sempre a mano alla recipe, anche in caso di elementi `input`, invece che affidarci alla pseudo-class